src/Diagnostics.DataProviders/DataProviders/GeoMasterDataProvider.cs

```diff
@@ -684,5 +685,43 @@ private string ParseGeoMasterName(string geomasterHostName)
 
             return geoMasterName;
         }
+
+        public override async Task<HealthCheckResult> CheckHealthAsync(CancellationToken cancellationToken = default)
+        {
+            HealthCheckResult result = null;
+
+            if (_configuration.HealthCheckInputs != null && _configuration.HealthCheckInputs.Any())
+            {
+                _configuration.HealthCheckInputs.TryGetValue("subscription", out string subscription);
+                _configuration.HealthCheckInputs.TryGetValue("resourceGroupName", out string resourceGroupName);
+                _configuration.HealthCheckInputs.TryGetValue("siteName", out string siteName);
+
+                var inputs = new string[] { subscription, resourceGroupName, siteName };
+
+                if (inputs.All(s => !string.IsNullOrWhiteSpace(s)))
+                {
+                    Exception geomasterException = null;
+                    try
+                    {
+                        var response = await this.GetAppSettings(subscription, resourceGroupName, siteName);
```

Comment:
I feel its more necessary now to secure the Health check endpoint as the Stakes are high.
We can potentially cause high geomaster CPU if someone Ddos the health check endpoint

Reply:
Also, The Appsettings call will go all the way to Stamp Database. So its very expensive.
for Health Check, maybe you want to consider a less expensive call which only involves geomaster.
I am little bit worried though about this call happening for every Health Check API request.

Another Design Model could be that you have a Service which pings these dependencies once every 5 mins (or X mins), logs the result in kusto and maintain the last Status Check result.
The API simply returns what was the last check result along with the timestamp.